src/_pytest/assertion/util.py

```diff
@@ -228,10 +230,32 @@ def _diff_text(left: str, right: str, verbose: int = 0) -> List[str]:
         explanation += ["Strings contain only whitespace, escaping them using repr()"]
     # "right" is the expected base against which we compare "left",
     # see https://github.com/pytest-dev/pytest/issues/3333
-    explanation += [
-        line.strip("\n")
-        for line in ndiff(right.splitlines(keepends), left.splitlines(keepends))
-    ]
+
+    from difflib import SequenceMatcher
+    stripped_left = ''.join(left.split())
```

Comment:
Why are you using `split()` (which will throw out whitespace) instead of `splitlines`? Shouldn't we consider whitespace, given that mismatched whitespace will appear on the diff?

Reply:
I thought whitespaces were less important. It make sense to consider whitespaces in SequenceMatcher.